src/rust-crypto/RoomEncryptor.ts

```diff
@@ -91,7 +103,33 @@ export class RoomEncryptor {
             );
         }
 
+        // Manually call `loadMembersIfNeeded` here, because we want to know if it's the first
+        // time the room is loaded (due to lazy loading), so we can update the tracked users.
+        const fromServer = await this.room.loadMembersIfNeeded();
         const members = await this.room.getEncryptionTargetMembers();
+
+        if (fromServer) {
+            // It's the first time the room is loaded, so we need to update the tracked users
+            await this.olmMachine
+                .updateTrackedUsers(members.map((u) => new RustSdkCryptoJs.UserId(u.userId)))
+                .then(() => {
+                    this.prefixedLogger.debug(`Updated tracked users for room ${this.room.roomId}`);
+                });
+        }
+
+        // Query keys in case we don't have them for newly tracked members.
```

Comment:
It's not really related to the current problem, but I had to fix it to make the lazy loading test pass on both crypto backend. It's anyhow a real problem, and we can see several [occurence of it in rageshakes](https://rageshake-search.tools.element.io/search_logs?log=some+devices+might+be+missing). The (full) rust sdk as a [similar mecanism](https://github.com/matrix-org/matrix-rust-sdk/blob/93edf7a06462c8ee9afcc587a5725aa8956ad5ab/crates/matrix-sdk/src/room/mod.rs#L1504) to ensure the download of user keys for non tracked/dirty users.

This would need to be fixed in mobile too